### ISSUE TITLE
Unused files check

### ISSUE THREAD

**Issue by madsenmm** [Shopify/theme-tools#518]
**Is your feature request related to a problem? Please describe.**
Would be a nice feature to allow theme check to find unused references, i.e. snippets, sections (_not sure if possible with JSON templates_), assets...

**Describe the solution you'd like**
Like the current [MissingTemplate](https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/missing-template) check, would be nice to have a check that detects files not used.

**Comment by karreiro** [comment]
Thank you @madsenmm! I've renamed the issue to avoid confusion with JSON templates. My understanding is that you're proposing a check for unused files (however, I don't think we have unused JSON templates).

**Comment by madsenmm** [comment]
Yes, more specifically snippets, just added sections and assets, in case it is possible to check somehow.
JSON templates won't be necessary, in the context I was thinking, was the check on sections, which probably going to be difficult, if not impossible when they can be used dynamically in JSON templates.

**Comment by madsenmm** [comment]
@charlespwd So not possible, or just way too CPU intensive task to do at runtime?

**Comment by charlespwd** [comment]
It's possible. But it would be CPU intensive if you wanted to run this check every time the file changes. You need to visit all the files, not just the open files to know if something is used or not. You also need to go across template.json files, sections/*.json files. 

In the CLI that's not much of a problem, since you're running theme check on all the files anyway. But in VS Code/language server, we only run theme check on the open files to keep things snappy (and avoid noise in the Problems window).

**Comment by madsenmm** [comment]
@charlespwd How would one go about only running via the CLI, and on on runtime, but more like only when running the `shopify theme check` command?
